src/main/java/io/aiven/connect/jdbc/sink/BufferedRecords.java

```diff
@@ -86,10 +90,15 @@ public List<SinkRecord> add(final SinkRecord record) throws SQLException {
         }
 
         final List<SinkRecord> flushed;
-        if (currentSchemaPair.equals(schemaPair)) {
+        // Skip the schemaPair check for all tombstone records or the current schema pair matches
+        if (record.valueSchema() == null || currentSchemaPair.equals(schemaPair)) {
             // Continue with current batch state
-            records.add(record);
-            if (records.size() >= config.batchSize) {
+            if (config.deleteEnabled && isTombstone(record)) {
+                tombstoneRecords.add(record);
+            } else {
+                records.add(record);
+            }
+            if (records.size() >= config.batchSize || tombstoneRecords.size() >= config.batchSize) {
                 log.debug("Flushing buffered records after exceeding configured batch size {}.",
```

Comment:
nit
might be worth adding in the records.size() and tombstoneRecords.size() so we know which one caused the flush

Reply:
Thanks @davidradl for the review, I have addressed most of the comments